src/__test__/graphql-flow.test.js

```diff
@@ -309,6 +331,105 @@ describe('graphql-flow generation', () => {
         });
     });
 
+    describe('Fragment dependency resolution', () => {
+        it('should do the one thing', () => {
+            const result = rawQueryToFlowTypes(
+                `query Hello {
+                    hero(episode: JEDI) {
+                        ...onChar
+                    }
+                }
+
+                fragment onChar on Character {
+                    __typename
+                    ... on Droid {
+                        primaryFunction
+                    }
+                }`,
+            );
+            expect(result).toMatchInlineSnapshot(`
+                // HelloType.js
+                export type HelloType = {|
+                    variables: {||},
+                    response: {|
+                  hero: ?({|
+                    __typename: "Droid",
+
+                    /** The robot's primary function*/
+                    primaryFunction: string,
+                  |} | {|
+                    __typename: "Human"
+                  |})
+                |}
+                |};
+
+                // onChar.js
+                export type onChar = {|
```

Comment:
hmmmm that might be tricky. There are cases where the type of the fragment is more general than the type that ends up in the implementation.
For example:
```
query Specified {
	human(id: "hello") {
		...onChar
	}
}
```
because it's being spread into a `Human` type, we know that it can't be a droid, and the type ends up being
```
type Specified = {
	hero: {
		__typename: "Human"
	}
}
```
(note the lack of primaryFunction).

The exported fragment type will always be a superset of the specialized type, however.

Reply:
oh hey I even test this in the next test case. I'll name them better 😅